pgmpy/base/DAG.py

```diff
@@ -1257,6 +1257,107 @@ def copy(self):
         dag.add_nodes_from(self.nodes())
         return dag
 
+    def edge_strength(self, data, edges=None):
+        """
+        Computes the strength of edges in the DAG using a residual-based conditional
+        independence test with Pillai's Trace effect size.
+
+        This method quantifies the dependence between variables after adjusting for
+        common parents, using a flexible approach that supports both continuous and
+        categorical data. It first residualizes variables using XGBoost (instead of
+        linear regression), then computes the canonical correlation between residuals,
+        and finally evaluates the strength using Pillai's Trace — a multivariate
+        effect size bounded between 0 and 1.
+
+        Pillai's Trace
+        --------------
+        - Uses a residualization-based approach (similar to partial correlation) and
+          employs the XGBoost estimator instead of linear regression.
+        - Computes the Pillai Trace effect size based on canonical correlations instead
+          of using correlation (as is the case with partial correlation).
+        - Measures the linear relationship between the residuals.
+        - Works for any mixture of categorical and continuous variables.
+        - The value is bounded between 0 and 1:
+        - Strength close to 1 → strong dependence.
+        - Strength close to 0 → conditional independence.
+
+        Parameters
+        ----------
+        data : pandas.DataFrame
+            Dataset to compute edge strengths on.
+
+        edges : tuple, list, or None (default: None)
+            - None: Compute for all DAG edges.
+            - Tuple (X, Y): Compute for edge X → Y.
+            - List of tuples: Compute for selected edges.
+
+        Returns
+        -------
+        dict
+            Dictionary mapping edges to their strength values.
+
+        Examples
+        --------
+        >>> import pandas as pd
+        >>> from pgmpy.base import DAG
+        >>> # Example DAG with edges and data
+        >>> dag = DAG([('X', 'Y'), ('Z', 'Y')])
+        >>> data = pd.DataFrame({'X': [...], 'Y': [...], 'Z': [...]})
+        >>> dag.edge_strength(data)
+        {('X', 'Y'): 0.99, ('Z', 'Y'): 0.98}
+
+        Notes
+        -----
+        - Based on the `ci_pillai` test using XGBoost-based residualization.
+        - Effect size is computed via Pillai's Trace on residuals' canonical correlations.
+
+        """
+
+        from pgmpy.estimators.CITests import pillai_trace
+
+        # If edges is None, compute for all edges in the DAG
+        if edges is None:
+            edges_to_compute = list(self.edges())
+        # If edges is a single edge tuple
+        elif isinstance(edges, tuple) and len(edges) == 2:
+            edges_to_compute = [edges]
+        # If edges is a list of edge tuples
+        else:
```

Comment:
I would suggest checking there that the argument here is a list of 2-tuples. And add another else, which throws an ValueError that the input by the user is not correct with some information on what we expect.